From c78aff56c9220ae98b969472468d51726b955103 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 16 Oct 2018 01:33:35 +0800
Subject: [PATCH 4875/5242] MLK-19931-2: dmaengine: fsl-edma: fix potential
 kernel crash in cyclic

commit  53d2794ec036f1d91f0a6da46890a0ab8cfddb4b from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix potential kernel crash issue  on fsl-edma as fsl-edma-v3.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Acked-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/fsl-edma.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/dma/fsl-edma.c b/drivers/dma/fsl-edma.c
index 023a331..afc4215 100644
--- a/drivers/dma/fsl-edma.c
+++ b/drivers/dma/fsl-edma.c
@@ -377,6 +377,7 @@ static int fsl_edma_terminate_all(struct dma_chan *chan)
 	fsl_edma_disable_request(fsl_chan);
 	fsl_chan->edesc = NULL;
 	fsl_chan->idle = true;
+	fsl_chan->vchan.cyclic = NULL;
 	vchan_get_all_descriptors(&fsl_chan->vchan, &head);
 	spin_unlock_irqrestore(&fsl_chan->vchan.lock, flags);
 	vchan_dma_desc_free_list(&fsl_chan->vchan, &head);
-- 
1.7.9.5

